test/MarketMaker.js

```diff
@@ -162,6 +215,69 @@ contract('marketMaker', function(accounts) {
     await help.waitToBlock(startBlock+blocksPerPeriod*numberOfMonth, accounts);
   };
 
+  it("Should return the correct sellPrice on every period", async function() {
+    token = await simulateCrowdsale(100, [40,30,20,10,0], accounts);
+
+    const startBlock = web3.eth.blockNumber+10;
+    const blocksPerPeriod = 5;
+    const initialPriceSpread = 100500;
+
+    mm = await LifMarketMaker.new(
+      token.address, startBlock, blocksPerPeriod, 24,
+      accounts[1], initialPriceSpread,
+      {value: web3.toWei(8, 'ether'), from: accounts[0]}
+    );
+
+    const initialSellPrice = parseInt(await mm.initialSellPrice());
+
+    await mm.calculateDistributionPeriods({from: accounts[4]});
+    await mm.calculateSellPricePeriods({from: accounts[3]});
+
+    help.debug('MM balance:', parseInt( web3.eth.getBalance(mm.address) ));
+    help.debug('Start block', parseInt( await mm.startBlock.call() ));
+    help.debug('Blocks per period', parseInt( await mm.blocksPerPeriod.call() ));
+    help.debug('Foundation address', await mm.foundationAddr.call() );
+    help.debug('Initial Wei', parseInt( await mm.initialWei.call() ));
+    help.debug('Initial Buy Price', parseInt( await mm.initialBuyPrice.call() ));
+    help.debug('Initial Sell Price', parseInt( await mm.initialSellPrice.call() ));
+
+    let accumIncrementPrice = [
+      0, 1000, 2010, 3030, 4060, 5101,
+      6152, 7213, 8285, 9368, 10462,
+      11566, 12682, 13809, 14947, 16096,
+      17257, 18430, 19614, 20810, 22019,
+      23239, 24471, 25716, 26973, 28243,
+      29525, 30820, 32129, 33450, 34784,
+      36132, 37494, 38869, 40257, 41660,
+      43076, 44507, 45952, 47412, 48886,
+      50375, 51878, 53397, 54931, 56481,
+      58045, 59626
+    ];
+
+    function parsePrice(price) {
+      return parseInt(price.toFixed(0)/10000);
+    }
+
+    await help.waitToBlock(startBlock+1);
+    for (var i = 0; i < 24; i++) {
+
+      // To log all the accumulative price incremenets in base 10^5
+      // help.debug((100000*(1.01**i))-100000 );
+
+      // help.debug(
+      //   parsePrice(parseFloat(initialSellPrice*(1.01**i)))
+      //   ,' == ',
+      //   parsePrice(parseFloat(await mm.getSellPrice()))
+      // );
+      assert.approximately(
+        parsePrice(parseFloat(initialSellPrice*(1.01**i))),
+        parsePrice(parseFloat(await mm.getSellPrice())),
+        1, 'wrong sell price in contract on period index'+ i
```

Comment:
The difference allowed here its not on the delta, its on the sellPrice received and the calculated outside the contract, and over the price in wei, 1 wei of difference between prices its a very high precision. Wen I format the price to Lif outside the contract the price its the same on the first 4 decimals.

```
Sell Price on period 0 :in memory 8.04000000000000032e-14  == on contract 8.04000000000000032e-14
Sell Price on period 1 :in memory 8.12039999999999966e-14  == on contract 8.12039999999999966e-14
Sell Price on period 2 :in memory 8.20160000000000011e-14  == on contract 8.20160000000000011e-14
Sell Price on period 3 :in memory 8.28360000000000042e-14  == on contract 8.28360000000000042e-14
Sell Price on period 4 :in memory 8.36640000000000058e-14  == on contract 8.36640000000000058e-14
Sell Price on period 5 :in memory 8.45010000000000026e-14  == on contract 8.45010000000000026e-14
Sell Price on period 6 :in memory 8.53459999999999980e-14  == on contract 8.53459999999999980e-14
Sell Price on period 7 :in memory 8.61990000000000046e-14  == on contract 8.61990000000000046e-14
Sell Price on period 8 :in memory 8.70609999999999937e-14  == on contract 8.70609999999999937e-14
Sell Price on period 9 :in memory 8.79320000000000034e-14  == on contract 8.79309999999999941e-14
Sell Price on period 10 :in memory 8.88109999999999989e-14  == on contract 8.88109999999999989e-14
Sell Price on period 11 :in memory 8.96990000000000023e-14  == on contract 8.96990000000000023e-14
Sell Price on period 12 :in memory 9.05960000000000009e-14  == on contract 9.05960000000000009e-14
Sell Price on period 13 :in memory 9.15019999999999947e-14  == on contract 9.15019999999999947e-14
Sell Price on period 14 :in memory 9.24169999999999964e-14  == on contract 9.24169999999999964e-14
Sell Price on period 15 :in memory 9.33410000000000059e-14  == on contract 9.33410000000000059e-14
Sell Price on period 16 :in memory 9.42749999999999946e-14  == on contract 9.42739999999999980e-14
Sell Price on period 17 :in memory 9.52180000000000038e-14  == on contract 9.52169999999999945e-14
Sell Price on period 18 :in memory 9.61699999999999956e-14  == on contract 9.61689999999999990e-14
Sell Price on period 19 :in memory 9.71309999999999953e-14  == on contract 9.71309999999999953e-14
Sell Price on period 20 :in memory 9.81029999999999961e-14  == on contract 9.81029999999999961e-14
Sell Price on period 21 :in memory 9.90840000000000047e-14  == on contract 9.90840000000000047e-14
Sell Price on period 22 :in memory 1.00075000000000005e-13  == on contract 1.00073999999999996e-13
Sell Price on period 23 :in memory 1.01075000000000001e-13  == on contract 1.01075000000000001e-13
```